lib/mumukit/explainer.rb

```diff
@@ -1,16 +1,21 @@
 class Mumukit::Explainer
   def explain(content, test_results)
     explain_methods
-        .map { |selector, key| eval_explain(selector, key, content, test_results) }
+        .flat_map { |selector, key| eval_explain(selector, key, content, test_results) }
         .compact
-        .map { |explain| I18n.t(explain[:key], explain[:binding]) }
-        .map { |it| "* #{it}" }
-        .join("\n")
+        .map do |key, binding|
+          if binding.key?(:type)
+            explanation = binding.merge message: I18n.t(key), line: binding[:line].to_i
```

Comment:
Those lines are a little hard to understand. Could you please extract some methods to improve readability?